tests/Integration/Behaviour/Features/Scenario/Product/Shop/delete_product_multishop.feature

```diff
@@ -0,0 +1,157 @@
+@reset-img-after-feature
+@clear-cache-after-feature
+@product-multishop
+@delete-product-multishop
+Feature: Delete product when multishop feature is enabled
+  As a BO user I want to be able to delete product from certain shops.
+
+  Background:
+    Given I enable multishop feature
+    And language with iso code "en" is the default one
+    And attribute group "Size" named "Size" in en language exists
+    And attribute group "Color" named "Color" in en language exists
+    And attribute "S" named "S" in en language exists
+    And attribute "M" named "M" in en language exists
+    And attribute "L" named "L" in en language exists
+    And attribute "White" named "White" in en language exists
+    And attribute "Black" named "Black" in en language exists
+    And attribute "Blue" named "Blue" in en language exists
+    And shop "shop1" with name "test_shop" exists
+    And shop group "default_shop_group" with name "Default" exists
+    And I add a shop "shop2" with name "test_second_shop" and color "red" for the group "default_shop_group"
+    And I add a shop group "test_second_shop_group" with name "Test second shop group" and color "green"
+    And I add a shop "shop3" with name "test_third_shop" and color "blue" for the group "test_second_shop_group"
+    And I add a shop "shop4" with name "test_shop_without_url" and color "blue" for the group "test_second_shop_group"
+    And single shop context is loaded
+    And I associate attribute group "Size" with shops "shop1,shop2,shop3,shop4"
+    And I associate attribute group "Color" with shops "shop1,shop2,shop3,shop4"
+    And I associate attribute "S" with shops "shop1,shop2,shop3,shop4"
+    And I associate attribute "M" with shops "shop1,shop2,shop3,shop4"
+    And I associate attribute "L" with shops "shop1,shop2,shop3,shop4"
+    And I associate attribute "White" with shops "shop1,shop2,shop3,shop4"
+    And I associate attribute "Black" with shops "shop1,shop2,shop3,shop4"
+    And I associate attribute "Blue" with shops "shop1,shop2,shop3,shop4"
+    And language "french" with locale "fr-FR" exists
+    And I add product "standardProduct" to shop "shop2" with following information:
+      | name[en-US] | magic staff |
+      | type        | standard    |
+    When I update product "standardProduct" stock for shop shop2 with following information:
+      | delta_quantity | 51 |
+    And I add new image "image1" named "app_icon.png" to product "standardProduct" for shop "shop2"
+    And I add new image "image2" named "some_image.jpg" to product "standardProduct" for shop "shop2"
+    When I set following shops for product "standardProduct":
+      | source shop | shop2             |
+      | shops       | shop1,shop3,shop4 |
+    And product standardProduct is associated to shop shop1
+    And product standardProduct is associated to shop shop2
+    And product standardProduct is associated to shop shop3
+    And product standardProduct is associated to shop shop4
```

Comment:
I think it would be worth changing the step sentence a bit to be able to use this way as well (optional though)
```suggestion
    And product standardProduct is associated to shops "shop1,shop2,shop3,shop4"
```